api/provider/aws/environment_delete.go

```diff
@@ -53,6 +59,42 @@ func (e *EnvironmentProvider) Delete(environmentID string) error {
 	return nil
 }
 
+func (e *EnvironmentProvider) checkEnvironmentDependencies(environmentID string) error {
+	checkDependentEntityTags := func(entityType string) error {
+		tags, err := e.TagStore.SelectByType(entityType)
+		if err != nil {
+			return err
+		}
+
+		dependentTags := tags.WithKey("environment_id").WithValue(environmentID)
+		if len(dependentTags) > 0 {
+			entityIDs := []string{}
+			for _, tag := range dependentTags {
+				entityIDs = append(entityIDs, tag.EntityID)
+			}
+
+			msg := fmt.Sprintf("Cannot delete environment '%s' because it contains dependent %ss:", environmentID, entityType)
+			msg += strings.Join(entityIDs, ", ")
+			return errors.Newf(errors.DependencyError, msg)
+		}
+		return nil
+	}
+
+	if err := checkDependentEntityTags("load_balancer"); err != nil {
```

Comment:
[nit] We could avoid some manual repetition here
```
for _, entity := range []string{"load_balancer", "service", "task"} {
    if err := checkDependentEntityTags(entity); err != nil {
        return err
    }
}
```

Reply:
I like